### ISSUE TITLE
Decouple tuning objective from evaluation and learning curve metrics

### ISSUE THREAD

**Issue by desilinguist** [EducationalTestingService/skll#350]
It would be nice to have separate ways to specify the metrics we want to use for evaluation (and for learning curve generation) that has nothing to do with tuning objective. Right now, those are linked and that's a little awkward when you aren't doing tuning at all or if you want to compute additional metrics too.

**Comment by aoifecahill** [comment]
Am I correct in understanding that `metrics` would be a list? That sounds good to me. 

For the learning curves does that then mean that you get multiple curves, one per metric?

**Comment by aloukina** [comment]
I like the `metrics` option. 
Perhaps, one can also raise a warning if the user specifies `objective` for models where no `grid_search` is done by definition such as OLS LR?
Another somewhat related suggestion: the default metric should probably be different for regressors and classifiers. Currently you get f1 if I am not mistaken.

**Comment by desilinguist** [comment]
@aloukina the last PR that's merged already [shows a warning](https://github.com/EducationalTestingService/skll/blob/master/skll/learner.py#L1182) if `grid_search` is `True` for a regressor for which no default param grid is available and no parameter grid is supplied. 

Changing the default is a bit trickier (given that we set the defaults for the configuration parser even before we have looked at the input config file) and should probably be a separate issue. However, the point is well taken and should be addressed. I will file a new issue.
